vault/vault.go

```diff
@@ -269,14 +269,8 @@ func NewTempCredentialsProvider(config *Config, keyring *CredentialKeyring) (aws
 	return t.GetProviderForProfile(config)
 }
 
-func NewFederationTokenCredentialsProvider(ctx context.Context, profileName string, k *CredentialKeyring, config *Config) (aws.CredentialsProvider, error) {
-	credentialsName, err := FindMasterCredentialsNameFor(profileName, k, config)
-	if err != nil {
-		return nil, err
-	}
-	masterCreds := NewMasterCredentialsProvider(k, credentialsName)
-
-	return NewFederationTokenProvider(ctx, masterCreds, config)
+func NewFederationTokenCredentialsProvider(ctx context.Context, profileName string, keyring *CredentialKeyring, config *Config) (aws.CredentialsProvider, error) {
+	return NewTempCredentialsProvider(config, keyring)
```

Comment:
A bit confused by this - what's the point of the `NewFederationTokenCredentialsProvider` function at all? You're not creating a federated token at all anymore

Reply:
Neither did the previous code. This is simply returning a credentials provider that shall be used to create the federation token.

The previous code went up the chain to the master credentials. This PR changes this to use temporary credentials as defined in the config.

Of course, we can also completely eliminate `NewFederationTokenCredentialsProvider` if you don't think there might be a need for different behaviour for creating federation tokens.

Reply:
Either way, found the right place to change to behaviour for configurations that use `credential_process`. Please see that latest commit.